draft-ietf-jsonpath-base.md

```diff
@@ -1463,19 +1470,21 @@ a type system is first introduced.
 
 ### Type System for Function Expressions {#typesys}
 
-Each parameter and the result of a function extension must have a declared type.
+Each parameter as well as the result of a function extension must have a declared type.
 
 A type is a set of instances.
 Declared types enable checking a JSONPath query for well-typedness
 independent of any argument the JSONPath query is applied to.
 
-{{tbl-types}} defines the available types in terms of abstract instances, where `n` denotes a node, `v` denotes a value, and `nl` denotes
+{{tbl-types}} defines the available types in terms of abstract instances, where `n` denotes a node, `v` denotes a value, `nl1` denotes a nodelist
+with at most one node, and `nl` denotes
 a nodelist.
 
 | Type                 | Abstract Instances                       |
 | :--                  | :----------------                        |
 | `ValueType`          | `Value(v)`, `Nothing`                    |
 | `LogicalType`        | `LogicalTrue`, `LogicalFalse`            |
+| `SingleNodeType`     | `Nodes(nl1)`                             |
```

Comment:
perhaps pragmatically name it `NodeType`.

Reply:
Perhaps. I did consider that but I thought it was too easy to confuse `NodeType` and `NodesType`. I also liked having some clue that the type is related to `singular-path`.

Reply:
hmm ... the reader is sensitised then and will pay attention to the difference.

But the relation to `singular-path` might be a good reason to keep the term of your choice.